widgetssdk/src/main/java/com/glia/widgets/entrywidget/EntryWidgetActivity.kt

```diff
@@ -1,19 +1,26 @@
 package com.glia.widgets.entrywidget
 
+import android.annotation.SuppressLint
 import android.os.Bundle
 import com.glia.widgets.base.FadeTransitionActivity
+import com.glia.widgets.di.Dependencies
 
 /**
  * EntryWidgetActivity provides a way to display the EntryWidget bottom sheet.
  */
 internal class EntryWidgetActivity : FadeTransitionActivity(), EntryWidgetFragment.OnDismissListener {
 
+    @SuppressLint("CheckResult")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
         if (savedInstanceState == null) {
             EntryWidgetFragment().show(supportFragmentManager)
         }
+
+        Dependencies.controllerFactory.entryWidgetHideController.onHide.subscribe {
```

Comment:
We also need to unsubscribe. We can have a memory leak here.

Reply:
I've made the changes, please have a look 3ca36afdfa7ae73da7c6c028729e56123fcc2235